vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/AbstractGridMultiSelectionModel.java

```diff
@@ -209,8 +209,10 @@ public void deselect(T item) {
     @Override
     public void selectAll() {
         updateSelection(
-                getGrid().getDataCommunicator().getDataProvider()
-                        .fetch(new Query<>()).collect(Collectors.toSet()),
+                (Set<T>) getGrid().getDataCommunicator().getDataProvider()
+                        .fetch(getGrid().getDataCommunicator().buildQuery(0,
```

Comment:
Let's add a test for this as well.

Reply:
Separated the test cases in the new commit.

Reply:
Sorry, I missed that they were combined, but extracting a separate test is an improvement 👍.